File: leetcode/lc-0009/solution.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (!x) return true;
        if (x < 0 || x % 10 == 0) return false;
        string s = to_string(x);
        int l = 0, r = s.size() - 1;
        while (l < r) {
            if (s[l] == s[r]) {
                l++;
                r--;
            } else {
                return false;
            }
        }
        return true;
    }
};

File: leetcode/lc-0009/solution.rs
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        // https://doc.rust-lang.org/stable/rust-by-example/flow_control/match/guard.html
        match x {
            0 => true,
            n if n < 0 || n % 10 == 0 => false,
            _ => {
                let s = x.to_string();
                let mut l = 0;
                let mut r = s.len() - 1;
                while l < r {
                    if s.as_bytes()[l] == s.as_bytes()[r] {
                        l += 1;
                        r -= 1;
                    } else {
                        return false;
                    }
                }
                true
            },
        }
    }
}
